### PR TITLE
feat(WMGV2): Improved method of adding genes to heatmap

### PR DESCRIPTION
**Pull request by seve**
## Reason for Change

- #4766

## Changes

- add
  - Improved method of adding genes to the heatmap 
- remove
  - Old QuickAdd code

## Notes for Reviewer

Note we now have three duplications of the same component, one in cell cards, one for cell type filter, and one for gene add

### CODE REVIEW

**Comment by seve** [comment]
lint errors are in unused (WMGV1) files, will address in the morning, but not required for review

File: frontend/src/components/common/Filter/descendant_mappings/cell_type_descendants.json
Comment (tihuan) [inline frontend/src/components/common/Filter/descendant_mappings/cell_type_descendants.json:1]: are these two json files related to this PR?

File: frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/index.tsx [hunk 1]
```diff
@@ -135,6 +137,9 @@ export default function XAxisChart({ geneNames }: Props): JSX.Element {
       left={Y_AXIS_CHART_WIDTH_PX + CHART_PADDING_PX}
       height={xAxisHeight}
     >
+      {!isSSR() && (
```
Comment (tihuan) [inline frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/index.tsx:140]: Let's add a note to remove this temp fix after https://github.com/chanzuckerberg/sci-components/pull/581 is published!

File: frontend/src/views/WheresMyGene/components/Main/index.tsx [hunk 2]
```diff
@@ -31,8 +31,6 @@ import { SideBarPositioner, SideBarWrapper, Top, Wrapper } from "../../style";
 import CellInfoBar from "../CellInfoSideBar";
```
Comment (tihuan) [inline frontend/src/views/WheresMyGene/components/Main/index.tsx:31]: should we just delete this v1 file?

File: frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/index.tsx [hunk 3]
```diff
@@ -100,19 +247,18 @@ export default function GeneSearchBar({
...
+  function handleSelectGenes(
+    _: unknown,
+    genes: DefaultAutocompleteOption[] | null,
+    __: unknown,
+    ___: unknown
+  ) {
     if (!dispatch) return;
 
-    dispatch(selectGenes(genes.map((gene) => gene.name)));
+    dispatch(selectGenes(genes?.map((gene) => gene.name) || []));
```
Comment (tihuan) [inline frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/index.tsx:262]:
```suggestion
    dispatch(selectGenes(genes?.map((gene) => gene.name) || EMPTY_ARRAY));
```

File: frontend/src/components/common/Filter/descendant_mappings/cell_type_descendants.json
Comment (seve) [reply frontend/src/components/common/Filter/descendant_mappings/cell_type_descendants.json:1]: no they were edited when running `npm run format`

File: frontend/ts_node_output.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Need to install the following packages:
```
[the remark this answers is not in the document]
Comment (seve) [reply frontend/ts_node_output.txt:1]: i have no idea what this is 🤔 removing

**Review by tihuan — COMMENTED** [review]
Code LGTM! I haven't had a chance to test the actual functionality, so hopefully someone else can help with that!

**Review by atarashansky — APPROVED** [review]

**Comment by tihuan** [comment]
@seve SDS 15.4.2 has the Autocomplete fix now, so maybe a followup PR to use it 🙏 ? Thank you!

**Pull request merged**